src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,675 @@
+use Generated\Shared\Transfer\OpenApiResponseTransfer;
+use SprykerSdk\Zed\AopSdk\AopSdkConfig;
+use Symfony\Component\Process\Process;
+
+class OpenApiCodeBuilder implements OpenApiCodeBuilderInterface
+{
+    /**
+     * @var string
+     */
+    protected const APPSRESOURCE = 'AppsResource';
+
+    /**
+     * @var string
+     */
+    protected const SPRYKER = 'Spryker';
+
+    /**
+     * @var string
+     */
+    protected const CORE = 'core';
+
+    /**
+     * @var \SprykerSdk\Zed\AopSdk\AopSdkConfig
+     */
+    protected AopSdkConfig $config;
+
+    /**
+     * @var string
+     */
+    protected string $sprykMode = 'project';
+
+    /**
+     * @var string
+     */
+    protected string $moduleName = '';
+
+    /**
+     * @var string
+     */
+    protected string $organization = '';
+
+    /**
+     * @var array
+     */
+    protected array $parsedData = [];
+
+    /**
+     * @param \SprykerSdk\Zed\AopSdk\AopSdkConfig $config
+     */
+    public function __construct(AopSdkConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OpenApiRequestTransfer $openApiRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\OpenApiResponseTransfer
+     */
+    public function build(OpenApiRequestTransfer $openApiRequestTransfer): OpenApiResponseTransfer
+    {
+        $openApiResponseTransfer = new OpenApiResponseTransfer();
+        $openApi = $this->load($openApiRequestTransfer->getTargetFileOrFail());
+
+        //Set default value for organization
+        $this->organization = $openApiRequestTransfer->getOrganizationOrFail();
+        if ($this->organization === static::SPRYKER) {
+            $this->sprykMode = static::CORE; //Set sprykMode based on organization
+        }
+
+        $openApiResponseTransfer = $this->buildCodeForOpenApi($openApi, $openApiResponseTransfer);
+
+        if ($openApiResponseTransfer->getMessages()->count() === 0) {
+            $messageTransfer = new MessageTransfer();
+            $messageTransfer->setMessage('Something went wrong. Either not channels have been found or the channels do not have messages defined.');
+            $openApiResponseTransfer->addError($messageTransfer);
+        }
+
+        return $openApiResponseTransfer;
+    }
+
+    /**
+     * @param string $openApiFilePath
+     *
+     * @return \cebe\openapi\spec\OpenApi
+     */
+    public function load(string $openApiFilePath): OpenApi
+    {
+        return Reader::readFromYamlFile((string)realpath($openApiFilePath));
+    }
+
+    /**
+     * @param string $resourceUri
+     *
+     * @return string
+     */
+    protected function getModuleName(string $resourceUri): string
+    {
+        $resources = explode('/', trim($resourceUri, '/'));
+
+        return ucwords(current($resources)) . 'Api';
+    }
+
+    /**
+     * @param string $resourceUri
+     *
+     * @return string
+     */
+    protected function getController(string $resourceUri): string
```

Comment:
Same as above let the method argument allow for more detailed checks against other parts of the definition.